torch/distributed/distributed_c10d.py

```diff
@@ -202,6 +202,13 @@ class Backend:
 
     backend_list = [UNDEFINED, GLOO, NCCL, UCC, MPI]
 
+    backend_capability: Dict[str, List[str]] = {
```

Comment:
You will also need to update this in the `register_backend` function (https://github.com/pytorch/pytorch/blob/73dac484641001d259c550966399ef28954a0a20/torch/distributed/distributed_c10d.py#L216) to update this dictionary to handle custom backends

Reply:
To keep BC, we should have register_backend register the custom backend to have both cpu and cuda capabilities

Reply:
Added these lines of code before seeing your comment:
https://github.com/pytorch/pytorch/pull/100238/files#diff-61109d1cb2a0bd13fc51d678a82666295289da1ec0a1a694e73d9e8c28f51bdcR298-R311
Would it work too?